quilt/tools/build.py

```diff
@@ -106,3 +120,94 @@ def build_package(username, package, yaml_path):
         _build_table(build_dir, store, '', tables)
         if readme is not None:
             _build_file(build_dir, store, 'README', rel_path=readme)
+
+def generate_build_file(startpath, outfilename='build.yml'):
+    startbase = os.path.basename(startpath)
+    buildfiles = {startbase : {}}
+    buildtables = {startbase : {}}
+
+    def add_to_buildfiles(path, files):
+        try:
+            safepath = [_pythonize_name(dir) if dir != '.' else '.' for dir in path]
+        except BuildException:
+            warning = "Warning: could not determine a Python-legal name for {path}; skipping."
+            print(warning.format(path=os.sep.join(path)))
+            return
+
+        ptr = buildfiles
+        for dir in path:
+            if dir not in ptr:
+                ptr[dir] = {}
+            ptr = ptr[dir]
+        for file in files:
+            fullpath = "/".join(path + [file])
+            try:
+                name, ext = file.split('.')
+            except ValueError:
+                # file with no extension
+                name = file
+            ptr[_pythonize_name(name)] = fullpath
+
+    def add_to_buildtables(path, files):
+        try:
+            safepath = [_pythonize_name(dir) if dir != '.' else '.' for dir in path]
+        except BuildException:
+            warning = "Warning: could not determine a Python-legal name for {path}; skipping."
+            print(warning.format(path=os.sep.join(path)))
+            return
+
+        ptr = buildtables
+        for folder in safepath:
+            if folder not in ptr:
+                # pythonize folder
+                ptr[folder] = {}
+            ptr = ptr[folder]
+        for file in files:
+            fullpath = "/".join(path + [file])
+            name, ext = file.split('.')
+            # pythonize name
+            ptr[_pythonize_name(name)] = [ext.lower(), fullpath]
+
+    for root, dirs, files in os.walk(startpath):
+        # skip hidden directories
+        for d in dirs:
+            if d.startswith('.'):
+                dirs.remove(d)
+
+        rel_path = os.path.relpath(root, startpath)
+        path = rel_path.split(os.sep)
```

Comment:
This might not work on Windows, where both "/" and "\" are separators. There's `os.path.split`, but it only splits once.